presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -210,18 +227,57 @@ public void addInput(Page page)
     @Override
     public Page getOutput()
     {
-        if (state != State.FINISHING || !blocked.isDone()) {
+        if (!blocked.isDone()) {
             return null;
         }
+
+        if (!statisticAggregation.isFinished()) {
+            Page aggregationOutput = statisticAggregation.getOutput();
+            if (aggregationOutput == null) {
+                return null;
+            }
+            int positionCount = aggregationOutput.getPositionCount();
+            Block[] outputBlocks = new Block[types.size()];
+            for (int channel = 0; channel < types.size(); channel++) {
+                if (channel == ROW_COUNT_CHANNEL || channel == FRAGMENT_CHANNEL) {
```

Comment:
These constants seem to be a false abstraction. We can't change them to be, say `3` and `4`, because that breaks the `channel - 2` below. So maybe it's better to just say `if (channel < 2)` here. Or add a new constant `WRITER_CHANNELS = 2`  and use that.